modules/core/src/lib/picking/pick-info.js

```diff
@@ -22,23 +22,32 @@
 // x, y, coordinates etc.
 export function getEmptyPickingInfo({pickInfo, mode, viewports, layerFilter, pixelRatio, x, y, z}) {
   const layer = pickInfo && pickInfo.pickedLayer;
-  const viewportFilter =
-    layerFilter &&
-    layer &&
-    (v =>
-      layerFilter({
-        layer,
-        viewport: v,
-        isPicking: true,
-        renderPass: `picking:${mode}`
-      }));
-  const viewport = getViewportFromCoordinates(viewports, {x, y}, viewportFilter);
-  const coordinate = viewport && viewport.unproject([x - viewport.x, y - viewport.y], {targetZ: z});
+
+  let pickedViewport = viewports[0];
+  if (viewports.length > 1) {
+    if (layerFilter && layer) {
+      let rootLayer = layer;
+      while (rootLayer.parent) rootLayer = rootLayer.parent;
```

Comment:
Nits:
- This filtering logic is a bit distributed. It would be nice it it was done in one place and this file called some helper.
- Our linter config doesn't require braces after `while` and related constructs? I was assuming it would.